openmc/deplete/results_list.py

```diff
@@ -296,3 +298,72 @@ def get_step_where(
+            Index of burnup step to evaluate
+        nuc_with_data : Iterable of str, optional
+            Nuclides iterable to evaluate with neutron data.
+            This must be specified if not all nuclides appearing in
+            depletion results have associated neutron cross sections, and
+            as such cannot be used in subsequent transport calculations.
+            If not provided, nuclides from the cross_sections element of
+            materials.xml will be used. If that element is not present,
+            nuclides from OPENMC_CROSS_SECTIONS will be used.
+
+        Returns
+        -------
+        mat_file : Materials
+            A modified Materials instance containing depleted material data,
+            plus original isotopic compositions of non-depletable materials
+        """
+        result = self[burnup_index]
+
+        # Only materials found in the original materials.xml file will be
+        # updated. If for some reason you have modified OpenMC to produce
+        # new materials as depletion takes place, this method will not
+        # work as expected and leave out that material.
+        mat_file = Materials.from_xml("materials.xml")
+
+        # Only nuclides with valid transport data will be written to
+        # the new materials XML file. The precedence of nuclides to select
+        # is first ones provided as a kwarg here, then ones specified
+        # in the materials.xml file if provided, then finally from
+        # the environment variable OPENMC_CROSS_SECTIONS.
+        if nuc_with_data:
+            if not all(isinstance(nuclide_name, str) for nuclide_name in nuc_with_data):
+                raise InvalidArgumentError("Expected an iterable of strings as acceptable nuclide data.")
+            available_cross_sections = nuc_with_data
+        else:
+            # select cross_sections.xml file to use
+            if mat_file.cross_sections:
+                this_library = DataLibrary.from_xml(path=mat_file.cross_sections)
+            else:
+                this_library = DataLibrary.from_xml()
+
+            # Find neutron libraries we have access to
+            available_cross_sections = set()
+            for lib in this_library.libraries:
+                if lib['type'] == 'neutron':
+                    available_cross_sections.update(lib['materials'])
+            if not available_cross_sections:
+                raise DataError('No neutron libraries found in cross_sections.xml')
+
+        # Overwrite material definitions, if they can be found in the depletion
+        # results, and save them to the new depleted xml file.
+        for mat in mat_file:
+            if str(mat.id) in result.mat_to_ind.keys():
+                mat.volume = result.volume[str(mat.id)]
+                for nuc in result.nuc_to_ind.keys():
+                    if nuc not in available_cross_sections:
+                        continue
+                    atoms = result[0, str(mat.id), nuc]
```

Comment:
```suggestion
            mat_id = str(mat.id)
            if mat_id in result.mat_to_ind:
                mat.volume = result.volume[mat_id]
                for nuc in result.nuc_to_ind:
                    if nuc not in available_cross_sections:
                        continue
                    atoms = result[0, mat_id, nuc]
```

Reply:
Good suggestion here. Was unaware iterating over dictionary keys was the default behavior.